diff -up blt2.5/generic/bltScrollbar.c.4~ blt2.5/generic/bltScrollbar.c
--- blt2.5/generic/bltScrollbar.c.4~	2010-12-26 18:16:50.000000000 +0100
+++ blt2.5/generic/bltScrollbar.c	2023-07-27 21:48:41.962595704 +0200
@@ -588,7 +588,7 @@ ScrollbarWidgetCmd(clientData, interp, a
 	} else {
 	    fraction = ((double)pixels / (double)barWidth);
 	}
-	sprintf(interp->result, "%g", fraction);
+	sprintf(Tcl_GetStringResult(interp), "%g", fraction);
     } else if ((c == 'f') && (strncmp(argv[1], "fraction", length) == 0)) {
 	int x, y, pos, barWidth;
 	double fraction;
@@ -887,7 +887,7 @@ ScrollbarImageProc(
  *
  * Results:
  *	The return value is a standard Tcl result.  If TCL_ERROR is
- *	returned, then interp->result contains an error message.
+ *	returned, then Tcl_GetStringResult(interp) contains an error message.
  *
  * Side effects:
  *	Configuration information, such as colors, border width,
diff -up blt2.5/generic/bltTed.c.4~ blt2.5/generic/bltTed.c
--- blt2.5/generic/bltTed.c.4~	2009-05-09 18:27:15.000000000 +0200
+++ blt2.5/generic/bltTed.c	2023-07-27 21:48:41.961595681 +0200
@@ -25,6 +25,8 @@
  * Table editor was created by George Howlett.
  */
 
+#define USE_INTERP_RESULT
+
 #include "bltInt.h"
 
 #include "bltTable.h"
diff -up blt2.5/generic/bltVecMath.c.4~ blt2.5/generic/bltVecMath.c
--- blt2.5/generic/bltVecMath.c.4~	2009-12-31 02:32:07.000000000 +0100
+++ blt2.5/generic/bltVecMath.c	2023-07-27 21:49:23.469550084 +0200
@@ -815,10 +815,10 @@ Fmod(x, y)
  *
  *	This procedure is called when an error occurs during a
  *	floating-point operation.  It reads errno and sets
- *	interp->result accordingly.
+ *	Tcl_GetStringResult(interp) accordingly.
  *
  * Results:
- *	Interp->result is set to hold an error message.
+ *	Tcl_GetStringResult(interp) is set to hold an error message.
  *
  * Side effects:
  *	None.
@@ -834,20 +834,20 @@ MathError(interp, value)
     if ((errno == EDOM) || (value != value)) {
 	Tcl_AppendResult(interp, "domain error: argument not in valid range",
 	    (char *)NULL);
-	Tcl_SetErrorCode(interp, "ARITH", "DOMAIN", interp->result,
+	Tcl_SetErrorCode(interp, "ARITH", "DOMAIN", Tcl_GetStringResult(interp),
 	    (char *)NULL);
     } else if ((errno == ERANGE) || IS_INF(value)) {
 	if (value == 0.0) {
 	    Tcl_AppendResult(interp, 
 			     "floating-point value too small to represent",
 		(char *)NULL);
-	    Tcl_SetErrorCode(interp, "ARITH", "UNDERFLOW", interp->result,
+	    Tcl_SetErrorCode(interp, "ARITH", "UNDERFLOW", Tcl_GetStringResult(interp),
 		(char *)NULL);
 	} else {
 	    Tcl_AppendResult(interp, 
 			     "floating-point value too large to represent",
 		(char *)NULL);
-	    Tcl_SetErrorCode(interp, "ARITH", "OVERFLOW", interp->result,
+	    Tcl_SetErrorCode(interp, "ARITH", "OVERFLOW", Tcl_GetStringResult(interp),
 		(char *)NULL);
 	}
     } else {
@@ -856,7 +856,7 @@ MathError(interp, value)
 	sprintf(buf, "%d", errno);
 	Tcl_AppendResult(interp, "unknown floating-point error, ",
 	    "errno = ", buf, (char *)NULL);
-	Tcl_SetErrorCode(interp, "ARITH", "UNKNOWN", interp->result,
+	Tcl_SetErrorCode(interp, "ARITH", "UNKNOWN", Tcl_GetStringResult(interp),
 	    (char *)NULL);
     }
 }
@@ -950,7 +950,7 @@ ParseString(interp, string, valuePtr)
  *	was computed successfully.  If the name doesn't match any
  *	known math function, returns TCL_RETURN. And if a format error
  *	was found, TCL_ERROR is returned and an error message is left
- *	in interp->result.
+ *	in Tcl_GetStringResult(interp).
  *
  *	After a successful return parsePtr will be updated to point to
  *	the character just after the function call, the token is set
@@ -1054,7 +1054,7 @@ ParseMathFunction(interp, start, parsePt
  * Results:
  *	TCL_OK is returned unless an error occurred while doing lexical
  *	analysis or executing an embedded command.  In that case a
- *	standard Tcl error is returned, using interp->result to hold
+ *	standard Tcl error is returned, using Tcl_GetStringResult(interp) to hold
  *	an error message.  In the event of a successful return, the token
  *	and field in parsePtr is updated to refer to the next symbol in
  *	the expression string, and the expr field is advanced past that
@@ -1313,7 +1313,7 @@ NextToken(interp, parsePtr, valuePtr)
  *
  * Results:
  *	Normally TCL_OK is returned.  The value of the expression is
- *	returned in *valuePtr.  If an error occurred, then interp->result
+ *	returned in *valuePtr.  If an error occurred, then Tcl_GetStringResult(interp)
  *	contains an error message and TCL_ERROR is returned.
  *	InfoPtr->token will be left pointing to the token AFTER the
  *	expression, and parsePtr->nextPtr will point to the character just
@@ -1895,7 +1895,7 @@ NextValue(interp, parsePtr, prec, valueP
  *
  * Results:
  *	The result is a standard Tcl return value.  If an error
- *	occurs then an error message is left in interp->result.
+ *	occurs then an error message is left in Tcl_GetStringResult(interp).
  *	The value of the expression is returned in *valuePtr, in
  *	whatever form it ends up in (could be string or integer
  *	or double).  Caller may need to convert result.  Caller
@@ -1963,7 +1963,7 @@ EvaluateExpression(interp, string, value
  * Results:
  *	Each procedure returns TCL_OK if it succeeds and places result
  *	information at *resultPtr.  If it fails it returns TCL_ERROR
- *	and leaves an error message in interp->result.
+ *	and leaves an error message in Tcl_GetStringResult(interp).
  *
  * Side effects:
  *	None.
@@ -2282,7 +2282,7 @@ EvalVectorExpr(interp, string, dataPtr,
  * Results:
  *	Each of the procedures below returns a standard Tcl result.
  *	If an error occurs then an error message is left in
- *	interp->result.  Otherwise the value of the expression,
+ *	Tcl_GetStringResult(interp).  Otherwise the value of the expression,
  *	in the appropriate form, is stored at *resultPtr.  If
  *	the expression had a result that was incompatible with the
  *	desired form then an error is returned.
@@ -2318,7 +2318,7 @@ Blt_ExprVector(interp, string, vecPtr)
 	Blt_VectorDuplicate(vPtr, nvPtr);
     } else {
 	register int i;
-	/* No result vector.  Put values in interp->result.  */
+	/* No result vector.  Put values in Tcl_GetStringResult(interp).  */
 	for (i = 0; i < nvPtr->length; i++) {
 	    string = Blt_Dtoa(interp, nvPtr->valueArr[i]);
 	    Tcl_AppendElement(interp, string);
diff -up blt2.5/generic/tkScrollbar.c.4~ blt2.5/generic/tkScrollbar.c
--- blt2.5/generic/tkScrollbar.c.4~	2010-12-26 18:20:20.000000000 +0100
+++ blt2.5/generic/tkScrollbar.c	2023-07-27 21:48:41.961595681 +0200
@@ -567,7 +567,7 @@ ScrollbarWidgetCmd(clientData, interp, a
 	} else {
 	    fraction = ((double)pixels / (double)barWidth);
 	}
-	sprintf(interp->result, "%g", fraction);
+	sprintf(Tcl_GetStringResult(interp), "%g", fraction);
     } else if ((c == 'f') && (strncmp(argv[1], "fraction", length) == 0)) {
 	int x, y, pos, barWidth;
 	double fraction;
@@ -866,7 +866,7 @@ ScrollbarImageProc(
  *
  * Results:
  *	The return value is a standard Tcl result.  If TCL_ERROR is
- *	returned, then interp->result contains an error message.
+ *	returned, then Tcl_GetStringResult(interp) contains an error message.
  *
  * Side effects:
  *	Configuration information, such as colors, border width,
